app/models/mixins/retirement_mixin.rb

```diff
@@ -129,15 +129,19 @@ def retire_now(requester = nil)
     if retired
       return if retired_validated?
       _log.info("#{retirement_object_title}: [#{name}], Retires On: [#{retires_on.strftime("%x %R %Z")}], was previously retired, but currently #{retired_invalid_reason}")
-    else
-      update_attributes(:retirement_requester => requester)
-      event_name = "request_#{retirement_event_prefix}_retire"
-      _log.info("calling #{event_name}")
-      begin
-        raise_retirement_event(event_name, requester)
-      rescue => err
-        _log.log_backtrace(err)
+    elsif retirement_state != "initializing"
```

Comment:
@tinaafitz doesn't this check also need to test for all the other states -- like retiring -- that a vm/service that is in the process of being retired could be?

Reply:
retiring or retired also needs check here

Reply:
`if retired` is already handled at the top of this condition.

Reply:
@d-m-u Right, as we discussed, "retiring" needs to be checked here. We should add a constant "initializing" with the  other retirement_states.

Reply:
@gmcculloug Yes, I agree that:
1. The ```retiring?``` method should include both the ```RETIRING``` and ```RETIREMENT_INITIALIZING``` retirement_states.
2. The ```retirement_check``` and ```start_retirement``` methods should check for ```retiring?```


I propose that we add one more method, ```retirement_initialized``` to check for the retirement_state of ```RETIREMENT_INITIALIZING``` for the start_retirement Automate method to check that the object has been initialized for retirement. The start_retirement Automate method currently checks for ```retired?``` and ```retiring?```. The ```retiring?``` check should be replaced with ```retirement_initialized``` method to prevent the retirement state machine from processing an object that was not "initialized" for retirement.(This could happen if someone directly called the retirement state machine without having gone through ```retire_now```.)